modules/core/src/lighting/light-source.js

```diff
@@ -0,0 +1,35 @@
+// default light source parameters
+const DEFAULT_LIGHT_POSITION = [0.0, 0.0, 1.0];
+const DEFAULT_LIGHT_DIRECTION = [0.0, 0.0, -1.0];
+const DEFAULT_LIGHT_INTENSITY = 1.0;
+const DEFAULT_LIGHT_COLOR = [255, 255, 255];
+
+class LightSource {
```

Comment:
Should apps be able to place light sources in a scenegraph (i.e. `LightSource extends Object3D`?

Doing so would allow for Point and Directional lights to be transformed/animated as part of a group of objects.

Reply:
Definitely we need our new scenegraph class can store light sources, if extends from Object3D is the only requirement, I will do it in this PR.

Reply:
>  if extends from Object3D is the only requirement, I will do it in this PR.

Yes. The one thing we will need to consider is that each `Object3D` holds a 4x4  model matrix. it contains position, scale, rotation for that node.

If you inherit from Object 3D, you could potentially use the modelmatrix to extract the position and directions for those light sources, instead of storing them separately.

But we can fine-tune that later.

Reply:
Done